app/controllers/characters.controller.js

```
@@ -0,0 +1,94 @@
+const Op = db.Sequelize.Op;
+const Characters = db.characters;
+
+// Create a new character
+exports.create = (req, res) => {
+  if (req.body.name === undefined) {
+    const error = new Error("Name for character can't be empty!");
+    error.statusCode = 400;
+    throw error;
+  }
+
+  const character = {
+    name: req.body.name
+  };
+
+  Characters.create(character)
+    .then((data) => {
+      res.send(data);
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message:
+          err.message ||
+          "Some error occurred while creating the character.",
+      });
+    });
+}
+
+// Find all characters in the database
+exports.findAll = (req, res) => {
+  const id = req.query.id;
+  var condition = id ? { id: { [Op.like]: `%${id}%`, }, } : null;
+
+  Characters.findAll({ where: condition })
+    .then((data) => {
+      res.send(data);
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message: err.message || "Some error occurred while retrieving character.",
+      });
+    });
+};
+
+exports.delete = (req, res) => {
+  const id = req.params.id;
+
+  Characters.destroy({
+    where: { id: id },
+  })
+    .then((number) => {
+      if (number == 1) {
+        res.send({
+          message: "characters was deleted successfully!",
+        });
+      } else {
+        res.send({
+          message: `Cannot delete characters with id=${id}. Maybe characters was not found!`,
+        });
+      }
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message: err.message || "Could not delete characters with id=" + id,
+      });
+    });
+};
+
+// Update a character by the id in the request
+exports.update = (req, res) => {
+  const id = req.params.id;
+
+  Characters.update(req.body, {
+    where: { id: id },
+  })
+    .then((number) => {
+      if (number == 1) {
+        res.send({
+          message: "Character was updated successfully.",
+        });
+      } else {
+        res.send({
```

should status maybe also be set to something like 404?  otherwise what is the likelihood that the caller is going to check the response body for a message whether or not the update API actually did the update successfully vs. silently failed because the id wasn't found

---

I don't think I understand this comment...

---

my guess is that in the frontend code where you call the backend you probably expect that if something goes wrong you'll get back a HTTP response whose status code is NOT 200, but the way this _was_ coded the response would be 200 but have a message in it saying something went wrong.  That almost certainly means the frontend would think everything is fine when in fact nothing was saved in the backend as expected.